hicommonwealth/commonwealth
# Create Community - Community Selection Step

## Link to Issue
Closes: #5646 #5633

## Description of Changes
- added content (community type buttons) to the `community type step` page 
- handled first part of the flow for non-signed in and signed in users that want to create a community
- added `new community admin` modal that displays user's addresses that are available for creating community

https://github.com/hicommonwealth/commonwealth/assets/14819225/e2835a77-46ce-401a-87f9-6928188cdbf3

## Test Plan
- this is just first part of the flow so it is not possible to test it properly
- some basic click around session might be performed:
  - add  `FLAG_NEW_CREATE_COMMUNITY=true` to `.env` file
  -  go to `/createCommunity` page
  - if you are not signed in:
    - after clicking one of the four community type you should be prompted with login modal
    - after successfully log in, you should see the modal with available addresses
    - from here you can either connect another wallet or continue to next step

  - if you are logged in:
    - after clicking one of the four community type you should be prompted with `new community admin` modal
    - in the dropdown you should see only addresses that are from selected chain base
    - from here you can either connect another wallet or continue to next step

- login options should be limited depending on community type you selected:
  -  evm/polygon: metamask / wallet connect
  - cosmos: leap / keplr
  - solana: phantom

## Deployment Plan

1. feature is hidden behind the feature flag `FLAG_NEW_CREATE_COMMUNITY` and it should not be available for now
